### PR TITLE
feat: display program version

### LINKED ISSUE #81: Please, add --version option
Self-explanatory.

Haskell devs can figure-out what version are they using from side sources, but users expect `--version` anyway now.

### PR DESCRIPTION
**Pull request by gbataille**
Fixes #81

Took me a while but I think I have found a clean way to do that, both with options-applicative to have an option that "stops the program", like `--help` and with cabal giving me the version number (https://www.haskell.org/cabal/users-guide/developing-packages.html#accessing-data-files-from-package-code)

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -84,7 +93,7 @@ opts =
 main :: IO ()
 main = do
   canUseColor <- supportsPretty
-  config <- Opt.execParser opts
+  config <- Opt.customExecParser (Opt.prefs Opt.showHelpOnError) opts
```
Comment (guibou) [inline app/Main.hs:96]: I took me a while to understand what you were doing here.

Apparently you want to display the help on error, which was not initially the case.

I feel that it is orthogonal to this commit. However That's a good idea, so why not.

**Review by guibou — APPROVED** [review]

File: app/Main.hs (hunk 1 above)
Comment (gbataille) [reply app/Main.hs:96]: Hum, might be a remnant of me trying to make it work. I agree that I think that this has nothing to do with the issue. I think I left that by mistake
